03_letters/n0npax/main_test.go

```diff
@@ -0,0 +1,109 @@
+package main
+
+import (
+	"bytes"
+	"strconv"
+	"testing"
+)
+
+func TestLetters(t *testing.T) {
+	testCases := map[string]struct {
+		input    string
+		expected map[rune]int
+	}{
+		"polish": {input: "dzień dobry",
+			expected: map[rune]int{'y': 1, ' ': 1, 'd': 2, 'z': 1, 'i': 1, 'e': 1, 'ń': 1, 'o': 1, 'b': 1, 'r': 1}},
+		"russian": {input: "доброе утро",
+			expected: map[rune]int{'о': 3, 'р': 2, 'т': 1, 'у': 1, ' ': 1, 'е': 1, 'б': 1, 'д': 1}},
+		"english": {input: "good morning",
+			expected: map[rune]int{'o': 3, 'g': 2, 'd': 1, 'm': 1, 'r': 1, 'n': 2, 'i': 1, ' ': 1}},
+		"chineese": {input: "早上好",
+			expected: map[rune]int{'早': 1, '上': 1, '好': 1}},
+	}
+
+	for name, test := range testCases {
+		t.Run(name, helperTestLetter(name, test.input, test.expected))
+	}
+}
+
+func helperTestLetter(name string, input string, expected map[rune]int) func(*testing.T) {
+	return func(t *testing.T) {
+		result := letters(input)
+		for k, v := range result {
+			if v != expected[k] {
+				for k := range result {
+					if expected[k] != result[k] {
```

Comment:
I think that this is starting to get quite confusing and difficult to read. I don't think it is necessary to do the final iteration over the result. Can you please remove the double nested loop like below?

```go
for k, count := range result {
  if _, ok := expected[k]; !ok {
    t.Errorf("Found unexpected key: %c", k)
  }
  if count != expected[k] {
    t.Errorf("Incorrect count for key %c. Found: %d, Expected %d", k, count, expected[k])
  }
}
```

Reply:
Thanks for that. Fixed :)